website/src/arcade/ArcadeEditor.tsx

```diff
@@ -15,128 +15,97 @@ import { ARCADE_STORAGE_KEYS } from "@site/src/arcade/consts";
-
-      if (!container || editor) return;
-
-      // Pull initial code
-      const storedCode = getStorageValue(ARCADE_STORAGE_KEYS.CODE);
-      if (storedCode) {
-        MODELS.ts.setValue(
-          lzstring.decompressFromEncodedURIComponent(storedCode)
-        );
-      }
-
-      monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
-        typeRoots: ["groqd", "zod"],
-        target: monaco.languages.typescript.ScriptTarget.ES5,
-        strict: true,
-        skipLibCheck: true,
-        esModuleInterop: true,
-      });
-
-      monaco.languages.typescript.typescriptDefaults.setExtraLibs(extraLibs);
-      monaco.languages.registerInlayHintsProvider(
-        "typescript",
-        createTwoslashInlayProvider()
+export const ArcadeEditor = ({ dispatch }: ArcadeEditorProps) => {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const editorRef = React.useRef<monaco.editor.IStandaloneCodeEditor>();
+  const activeModel = "ts";
+
+  /**
+   * Set up editor on mount
+   */
+  React.useEffect(() => {
+    const container = containerRef.current;
+    let editor = editorRef.current;
+
+    if (!container || editor) return;
+
+    // Pull initial code
+    const storedCode = getStorageValue(ARCADE_STORAGE_KEYS.CODE);
+    if (storedCode) {
+      MODELS.ts.setValue(
+        lzstring.decompressFromEncodedURIComponent(storedCode)
       );
+    }
 
-      const handleContentChange = debounce(
-        () => runCode({ editor, dispatch }),
-        500
-      );
-      const didChangeInstance =
-        MODELS.ts.onDidChangeContent(handleContentChange);
+    monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
+      typeRoots: ["groqd", "zod"],
+      target: monaco.languages.typescript.ScriptTarget.ES5,
+      strict: true,
+      skipLibCheck: true,
+      esModuleInterop: true,
+    });
 
-      editorRef.current = monaco.editor.create(container, {
-        model: MODELS[activeModel],
-        language: "ts",
-        fontSize: 13,
-        automaticLayout: true,
-        minimap: { enabled: false },
-      });
-      editor = editorRef.current;
+    monaco.languages.typescript.typescriptDefaults.setExtraLibs(extraLibs);
+    monaco.languages.registerInlayHintsProvider(
+      "typescript",
+      createTwoslashInlayProvider()
+    );
 
-      // Cmd + Enter to run query
-      editor.addAction({
-        id: "trigger-run-query",
-        label: "Trigger Arcard query run",
-        keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.Enter],
-        run() {
-          runCode({ editor, dispatch, shouldRunQueryImmediately: true });
-        },
-      });
+    const handleContentChange = debounce(
+      () => runCode({ editor, dispatch }),
+      500
+    );
+    const didChangeInstance = MODELS.ts.onDidChangeContent(handleContentChange);
+
+    editorRef.current = monaco.editor.create(container, {
+      model: MODELS[activeModel],
+      language: "ts",
+      fontSize: 13,
```

Comment:
in the future might be neat to have a set of options that get saved in local storage like font size and a small collection of different font families.

Reply:
oh damn, that'd be really neat! noted! (I'd love to have a slick editor font, like fira code or jetbrains mono or something like that.)